zoom_rooms/indico_zoom_rooms/plugin.py

```diff
@@ -0,0 +1,271 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2024 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+import typing as t
+from functools import wraps
+
+from flask_pluginengine import render_plugin_template
+from wtforms.fields import BooleanField, FloatField, URLField
+from wtforms.validators import URL, DataRequired, NumberRange
+
+from indico.core import signals
+from indico.core.plugins import IndicoPlugin, PluginCategory
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.models.events import Event
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.events.timetable.models.entries import TimetableEntry
+from indico.modules.vc.models.vc_rooms import VCRoom, VCRoomEventAssociation
+from indico.web.forms.base import IndicoForm
+from indico.web.forms.fields import IndicoPasswordField
+from indico.web.forms.widgets import SwitchWidget
+
+from indico_zoom_rooms import _
+from indico_zoom_rooms.models import OperationArgs, ZoomRoomsAction, ZoomRoomsQueueEntry
+from indico_zoom_rooms.util import get_vc_room_associations, get_zoom_room_id
+
+
+class PluginSettingsForm(IndicoForm):
+    debug = BooleanField(
+        _('Debug mode'),
+        widget=SwitchWidget(),
+        description=_('If enabled, requests are not sent to the API but logged instead'),
+    )
+    service_url = URLField(
+        _('Service URL'), [URL(require_tld=False)], description=_('The URL of the CERN calendar service')
+    )
+    docs_url = URLField(_('Documentation URL'), [URL()], description=_('The URL the "Zoom Rooms" buttons will link to'))
```

Comment:
This should either be `DataRequired()` to show up as required (the `URL()` validator does not allow empty URLs, but it's weird to not have the field be shown as required), or `Optional()`; in the latter case the button should only be shown if there's a URL.